include/RAJA/pattern/kernel/ArgHelper.hpp

```diff
@@ -0,0 +1,179 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   Header file for kernel lambda executor.
+ *
+ ******************************************************************************
+ */
+
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-19, Lawrence Livermore National Security, LLC.
+//
+// Produced at the Lawrence Livermore National Laboratory
+//
+// LLNL-CODE-689114
+//
+// All rights reserved.
+//
+// This file is part of RAJA.
+//
+// For details about use and distribution, please read RAJA/LICENSE.
+//
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+
+#ifndef RAJA_pattern_kernel_arghelper_HPP
+#define RAJA_pattern_kernel_arghelper_HPP
+
+#include "RAJA/config.hpp"
+
+#include <iostream>
+#include <type_traits>
+
+#include "camp/camp.hpp"
+#include "camp/concepts.hpp"
+#include "camp/tuple.hpp"
+
+#include "RAJA/util/macros.hpp"
+#include "RAJA/util/types.hpp"
+
+#include "RAJA/pattern/kernel/Param.hpp"
+#include "RAJA/pattern/kernel/LambdaArgs.hpp"
+
+namespace RAJA
+{
+
+namespace internal
+{
+
+using RAJA::statement::seg_t;
+using RAJA::statement::param_t;
+using RAJA::statement::offset_t;
+using RAJA::statement::LambdaArgs;
+
+//Concatanate lists
+template<typename ListA, typename ListB>
+struct merge_list
+{};
+
+template<typename...itemsA, typename...itemsB>
+struct merge_list<camp::list<itemsA...>, camp::list<itemsB...>>
+{
+  using type = typename camp::list<itemsA...,itemsB...>;
+};
+
+template<typename...itemsA>
+struct merge_list<camp::list<itemsA...>, camp::list<>>
+{
+  using type  = typename camp::list<itemsA... >;
+};
+
+//List Maker
+template<typename Arg>
+struct listMaker
+{
+  using type = typename camp::list<>;
+};
+  
+//Convert LambdaArgs<T, 1, 2, 3> - > camp::list<LambdaArgs<T, 1>, LambdaArgs<T, 2>, LambdaArgs<T, 3> >
+template<typename T, camp::idx_t head, camp::idx_t... tail>
+struct listMaker<LambdaArgs<T, head, tail...>>
+{
+  using type = typename merge_list<camp::list<LambdaArgs<T, head>>,
```

Comment:
The additional types/expansions generated will make compilers cry. The following should make them happier:

```cpp
template <typename T, camp::idx_t... elems>
struct listMaker<LambdaArgs<T, elems...>>
{
  using type = camp::list<LambdaArgs<T, elems>...>;
};
```

Note: this also removes the need for the specialization below